src/app/designationEditor/designationEditor.component.js

```diff
@@ -0,0 +1,230 @@
+import angular from 'angular';
+import 'angular-sanitize';
+import 'angular-ui-bootstrap';
+import find from 'lodash/find';
+
+import commonModule from 'common/common.module';
+import sessionEnforcerService, {EnforcerCallbacks, EnforcerModes} from 'common/services/session/sessionEnforcer.service';
+import sessionService, {Roles} from 'common/services/session/session.service';
+import loadingOverlay from 'common/components/loadingOverlay/loadingOverlay.component';
+
+import titleModalController from './titleModal/title.modal';
+import pageOptionsModalController from './pageOptionsModal/pageOptions.modal';
+import photoModalController from './photoModal/photo.modal';
+import textEditorModalController from './textEditorModal/textEditor.modal';
+import websiteModalController from './websiteModal/website.modal';
+
+import template from './designationEditor.tpl';
+import titleModalTemplate from './titleModal/titleModal.tpl';
+import pageOptionsModalTemplate from './pageOptionsModal/pageOptionsModal.tpl';
+import photoModalTemplate from './photoModal/photoModal.tpl';
+import textEditorModalTemplate from './textEditorModal/textEditorModal.tpl';
+import websiteModalTemplate from './websiteModal/websiteModal.tpl';
+
+import designationConstants from './designationEditor.constants';
+
+let componentName = 'designationEditor';
+
+class DesignationEditorController {
+
+  /* @ngInject */
+  constructor( $http, $q, $uibModal, $location, $window, envService, sessionService, sessionEnforcerService ) {
+    this.sessionService = sessionService;
+    this.sessionEnforcerService = sessionEnforcerService;
+    this.imgDomain = envService.read('imgDomain');
+    this.imgDomainDesignation = envService.read('imgDomainDesignation');
+    this.$location = $location;
+    this.$http = $http;
```

Comment:
I'd suggest whatever is using $http get moved to an api service and out of the controller.